crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -564,9 +564,22 @@ impl<'w> EntityMut<'w> {
     /// # assert_eq!(new_r.0, 1);
     /// ```
     pub fn world_scope<U>(&mut self, f: impl FnOnce(&mut World) -> U) -> U {
-        let val = f(self.world);
-        self.update_location();
-        val
+        // Erase the lifetime of self, so we can get around the borrow checker.
+        let this_ptr = self as *mut Self;
```

Comment:
would it be worth creating a WorldScope struct to hold the drop impl and avoid the unsafe?